### PR TITLE
Upgrade operator-sdk to 0.11.0.

### PR DESCRIPTION
**Pull request by timmyers**
# Changes

- Upgrade the operator-sdk to 0.11.0 following https://github.com/operator-framework/operator-sdk/blob/master/doc/migration/version-upgrade-guide.md.
- This will fix the issue reference in this comment:
```
// NOTE: need to use a deepcopy since Status().Update() seems to reset the
// APIVersion of the res to "" making the object invalid; may be a mechanism
// to prevent us from using stale version of the object
```
and ultimately allow fixing a major bug related to expecting the cached reader to give up to date results.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

### CODE REVIEW

**Comment by timmyers** [comment]
These diffs are currently very large due to the `vendor` directory, what is the reason for it being in github?

**Comment by vdemeester** [comment]
> These diffs are currently very large due to the `vendor` directory, what is the reason for it being in github?

Mainly to be able to compile using older go version (if needed / if we can), and also because it reduces the time to build, etc.. (aka we don't download the deps on each and every CI run, …).
It not a huge deal for reviews on GitHub as it defaults to hide the vendor folder :wink:

**Comment by sthaha** [comment]
@timmyers thank you for the patch. Could you please address the integration-test failure? 🙏

File: pkg/apis/operator/v1alpha1/addon_types.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ type AddonCondition struct {
 
 // Addon is the Schema for the addons API
 // +k8s:openapi-gen=true
-// +kubebuilder:subresource:status
+// +kubebuilder:resource:path=addons,scope=Namespaced
```
Comment (sthaha) [inline pkg/apis/operator/v1alpha1/addon_types.go:41]: aren't the addons cluster-scoped?

Comment (timmyers) [reply pkg/apis/operator/v1alpha1/addon_types.go:41]: Sorry this should be reverted, it was an attempt to fix the bug that did not do what I expected.

**Comment by timmyers** [comment]
> Could you please refactor these into 2 PR 
Sure, I will update this PR to just bump the sdk version since its a prerequisite for fixing the bug.

**Pull request merged**